packages/worker/src/tests/TestConfiguration.ts

```diff
@@ -20,22 +20,45 @@ import {
   auth,
   constants,
   env as coreEnv,
+  utils,
+  DEFAULT_TENANT_ID,
```

Comment:
It seems no longer safe to use `DEFAULT_TENANT_ID` and `TENANT_ID` constants from below, they have the value of `default` and will be inherently racey. 

Do you think it would be possible to use an autogenerated tenant id instead like we are using for the `#tenantId` here? That might help with the `retry` logic required in this file.

Reply:
I think this is required as we are using "default" by default in the code itself. If we put something random, all the pieces of code that run `if tenant == 'default'` would fail. And this is still needed for tests such as migrations, plugins... global stuff like that

Reply:
In general we only check for `default` when `MULTI_TENANT` is false, and use that as a shortcut instead of accessing the tenant id in context. See: https://github.com/Budibase/budibase/blob/develop/packages/backend-core/src/context/mainContext.ts#L172-L174

As the worker has multi tenant set to true it should be safe, and it should save a lot of trouble with conflicting tests going forward. Each tenant would have it's own migrations and plugins in their respective global-db then.

Reply:
Some tests required multi_tenant to be false, as they were part of the test. I did not want to change the logic of these tests. Maybe worth to note it down and tackling it separately, as this was already like this, https://github.com/Budibase/budibase/blob/develop/packages/worker/src/tests/TestConfiguration.ts#L155

Reply:
I think some tests require there to be `SELF_HOST`, which should be a fine configuration we can allow for a self hosted multi tenant test. 

Do you know which ones require single tenancy? It shouldn't be the case.

Given that worker runs tests as `MULTI_TENANCY=1`, using the constant `default` as the tenant id in all tests undoes a lot of the benefit. Changing it to a random should help us a lot with conflicting test state.

Reply:
We have the helper `modeCloud` that was used in multiple places
 https://github.com/Budibase/budibase/blob/feature/dependencies-image/packages/worker/src/tests/TestConfiguration.ts#L101